### PR TITLE
feat(slider): add single tap gesture to slider

### PR DESCRIPTION
**Pull request by Hooolm**
First of all, love your work!

I would suggest setting this threshold to 0, or make it configurable. This make the controls of lights,covers, colors etc more snappy. And you do not have to pull it to change it rather than tapping it to change it.

### CODE REVIEW

**Comment by Hooolm** [comment]
I found that this is in fact not even enough. Will return with a more elaborate PR.

**Comment by piitaya** [comment]
It's an advanced usage. May be we can add a variable in the mushroom theme and get it in JS. I will give it some try.
The reason we choose 10 is for usability on mobile device to avoid trigger slider when scroll in iOS for example.

**Comment by Hooolm** [comment]
Yeah i agree that the threshold makes sense. I just overlooked the tap in hammer.js.

Consider adding this in.

File: src/shared/slider.ts [hunk 1]
```diff
@@ -104,6 +107,18 @@ export class SliderItem extends LitElement {
...
+                const percentage = getPercentageFromEvent(e);
+                this.value = this.percentageToValue(percentage);
+                this.dispatchEvent(
+                    new CustomEvent("current-change", {
+                        detail: {
+                            value: Math.round(this.value),
+                        },
+                    })
+                );
+            });
```
Comment (piitaya) [inline src/shared/slider.ts:121]: I think a dispatch change event is missing here (like in the move end event)
```
this.dispatchEvent(
    new CustomEvent("change", {
        detail: {
            value: Math.round(this.value),
        },
    })
);
```

**Review by piitaya — CHANGES_REQUESTED** [review]

**Comment by piitaya** [comment]
I just pushed another PR to add threshold customization if needed #219

File: src/shared/slider.ts (hunk 1 above)
Comment (Hooolm) [reply src/shared/slider.ts]: I agree and that should do it. Did not see that the difference in change and current-change.

**Comment by Hooolm** [comment]
I will look into the threshold config

**Comment by Hooolm** [comment]
Not sure how to test with the new config, other than the default value works, and tap works!

So in my eyes they work more snappy now without changing existing "swipe-gaurds".

**Comment by piitaya** [comment]
Thanks for the changes. No need to merge my branch into yours. I will merge the separate two PR 🙂

**Pull request merged**